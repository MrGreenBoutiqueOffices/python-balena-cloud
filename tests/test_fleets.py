"""Test the fleets model for Balena Cloud."""

# pylint: disable=too-many-arguments,too-many-positional-arguments
from __future__ import annotations

from typing import TYPE_CHECKING, Any

import pytest
from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from balena_cloud import BalenaCloud


async def test_get_fleets(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the get_fleets method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/application",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("fleets/all_fleets.json"),
        ),
    )
    fleets = await balena_cloud_client.get_fleets()
    assert fleets == snapshot


@pytest.mark.parametrize(
    ("param_key", "param_value", "endpoint"),
    [
        ("fleet_id", 1, "/v7/application(1)"),
        ("fleet_slug", "test-slug", "/v7/application(slug='test-slug')"),
        ("fleet_name", "Test Fleet", "/v7/application"),
    ],
)
async def test_get_fleet(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
    param_key: str,
    param_value: Any,
    endpoint: str,
) -> None:
    """Test the get_fleet method."""
    aresponses.add(
        "api.balena-cloud.com",
        endpoint,
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("fleets/fleet.json"),
        ),
    )
    # Get the fleet based on the parameter key
    fleet = None
    if param_key == "fleet_id":
        fleet = await balena_cloud_client.get_fleet(fleet_id=param_value)
    elif param_key == "fleet_slug":
        fleet = await balena_cloud_client.get_fleet(fleet_slug=param_value)
    elif param_key == "fleet_name":
        fleet = await balena_cloud_client.get_fleet(fleet_name=param_value)
    assert fleet == snapshot


async def test_get_fleet_devices(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the get_fleet_devices method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/device",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("fleets/fleet_devices.json"),
        ),
    )
    devices = await balena_cloud_client.get_fleet_devices(fleet_id=1)
    assert devices == snapshot


async def test_get_organization_fleets(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    balena_cloud_client: BalenaCloud,
) -> None:
    """Test the get_organization_fleets method."""
    aresponses.add(
        "api.balena-cloud.com",
        "/v7/application",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("fleets/all_fleets.json"),
        ),
    )
    fleets = await balena_cloud_client.get_organization_fleets(org_handle="test-org")
    assert fleets == snapshot
