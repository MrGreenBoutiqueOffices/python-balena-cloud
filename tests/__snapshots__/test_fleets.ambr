# serializer version: 1
# name: test_get_fleet[fleet_id-1-/v7/application(1)]
  Fleet(id=1, name='app_1', slug='organization/app_1', uuid='00000000000000000000000000000000', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True)
# ---
# name: test_get_fleet[fleet_name-Test Fleet-/v7/application]
  Fleet(id=1, name='app_1', slug='organization/app_1', uuid='00000000000000000000000000000000', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True)
# ---
# name: test_get_fleet[fleet_slug-test-slug-/v7/application(slug='test-slug')]
  Fleet(id=1, name='app_1', slug='organization/app_1', uuid='00000000000000000000000000000000', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True)
# ---
# name: test_get_fleet_devices
  list([
    Device(id=1, name='Device_1', status='Idle', uuid='00000000000000000000000000000001', is_online=True, is_web_accessible=True, is_undervolted=False, latitude=0.0, longitude=0.0),
    Device(id=2, name='Device_2', status='Idle', uuid='00000000000000000000000000000002', is_online=False, is_web_accessible=False, is_undervolted=False, latitude=1.0, longitude=1.0),
  ])
# ---
# name: test_get_fleets
  list([
    Fleet(id=1, name='app_1', slug='organization/app_1', uuid='00000000000000000000000000000000', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True),
    Fleet(id=2, name='app_2', slug='organization/app_2', uuid='11111111111111111111111111111111', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True),
  ])
# ---
# name: test_get_organization_fleets
  list([
    Fleet(id=1, name='app_1', slug='organization/app_1', uuid='00000000000000000000000000000000', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True),
    Fleet(id=2, name='app_2', slug='organization/app_2', uuid='11111111111111111111111111111111', created_at=datetime.datetime(2024, 1, 1, 0, 0, tzinfo=datetime.timezone.utc), is_public=False, is_host=False, is_archived=False, is_discoverable=True),
  ])
# ---
